mne/decoding/tests/test_csp.py

```diff
@@ -271,7 +271,7 @@ def test_ajd():
     A /= np.atleast_2d(np.sqrt(np.sum(A ** 2, 1))).T
     covmats = np.empty((n_times, n_channels, n_channels))
     for i in range(n_times):
-        covmats[i] = np.dot(np.dot(A, np.diag(diags[i])), A.T)
+        covmats[i] = np.linalg.multi_dot([A, np.diag(diags[i]), A.T])
```

Comment:
If we're going to optimize for performance anyway, might as well do this
```suggestion
        np.linalg.multi_dot([A, np.diag(diags[i]), A.T], out=convmats[i])
```

Reply:
Those requirements should be satisfied. At worst you'll need to add a dtype to the empty call but I doubt it